woocommerce/woocommerce-ios
# Order Fulfillment: Allow the user to delete package tracking details

Implements #851 

This PR brings back the UI to delete a shipment tracking, removed during the review of PR #817 to reduce the scope.

The big issue here is that I haven't been able to make deletion work. On the bright side, running the branch as it is and trying to delete a shipment tracking will help test that the error notice is presented.

![Simulator Screen Shot - iPhone 8 - 2019-04-10 at 11 59 54](https://user-images.githubusercontent.com/2722505/55851126-fd4f7180-5b89-11e9-9937-97cef00992f5.png)


## Changes:
- Minor changes to `ManualTrackingViewModel` and its two implementations (and its tests) to expose the style of the accessory view in the top cell.
- Minor changes to `FulfillViewController` to bring back the code that launches the UI
- Some changes to `ManualTrackingViewController` to remove the spinner when an action (save or delete) fails

The functionality is somehow more limited that what the original design suggested. As discussed, the backend API does not seem to support updating trackings, so we only allow users to delete them.

## Testing
- Checkout the branch, run the tests
- Navigate to an order with shipment trackings > Fullfill order
- Tap a shipment tracking
- Tap "Delete Tracking". At this moment, that should show an error notice (as mentioned, deleting does not actually work right now)

@bummytime @mindgraffiti could you please 👀 ? 

Also, could you please take a look at the code that deletes the tracking (that'd be `ShipmentsRemote.deleteShipmentTracking`) I assume there is something obvious that I am missing but I just can't see it.

Update release notes:
